### DISCUSSION TITLE
Add a function request similar to Algolia's Rules

### DISCUSSION THREAD

**Discussion by Jengro777** [meilisearch/product#546] [Feedback & Feature Proposal]
![image](https://user-images.githubusercontent.com/35259040/194832648-fd9267d3-ed9a-41e8-a114-0bb751ba3c32.png)
Refer to：https://www.algolia.com/doc/guides/managing-results/rules/rules-overview/

It is very useful for scenarios that need to be billed through search to fix the search sorting position of documents or improve the sorting position of certain types of documents according to search criteria

**Comment by Jengro777** [comment]
We have proposed this function request many times. Do Meilisearch have plans in the future? I'm looking forward to it

**Reply by gmourier** [reply]
Hey @Avey777 👋

We have set other priorities to solve before. We will probably work on it in the future because we also see a great addition to it.

When we prioritize it and start exploring it I will update everyone here. In the meantime, we can start the discussion and see what could be needed and for which use case.

Could you describe precise scenarios where it's needed for your use case?

Thanks

**Reply by Jengro777** [reply]
We are an e-commerce platform. Our mall provides upstream users with services that can fix a certain commodity and improve a certain category, and allows our upstream users to pay for it. This is very important for our revenue. A long time ago, we found that the cost of Algolia we use is getting higher and higher. We need to know that we will continue to pay more for Algolia in more countries. I have been following Meilishearh for nearly two years. At present, there are two reasons that prevent us from migrating to Meilishearch:

1. Personalization

2. Rules function that can fix and improve the sorting of search categories (refer to Algolia

**Reply by Jengro777** [reply]
> Hey @Avey777 wave
> 
> We have set other priorities to solve before. We will probably work on it in the future because we also see a great addition to it.
> 
> When we prioritize it and start exploring it I will update everyone here. In the meantime, we can start the discussion and see what could be needed and for which use case.
> 
> Could you describe precise scenarios where it's needed for your use case?
> 
> Thanks

**Reply by Jengro777** [reply]
It has been a long time. I want to know whether this function has been planned?

**Reply by gmourier** [reply]
Hello @Avey777 👋 

Sorry for the delay and thank you for your previous answers. As you can see, the feature receives very little interest for the moment from the community. We haven't had time to look into it, but I've noted it as something to explore when we have more time. We've been pretty busy with the v1.0 release.

When something progresses on a topic, people are usually notified in the relative discussion if there is one. That is, I'll let you know when we will start to explore it.

Thanks!

**Comment by Jengro777** [comment]
![image](https://user-images.githubusercontent.com/35259040/195758913-369812bd-ca3a-4670-a468-4b8eacd25b97.png)
I hope this function request can be placed in the roadmap so that you can see it more clearly

**Comment by Amie777** [comment]
Specifying the display location of items during search is a very useful function for e-commerce platforms. Now that the 1.0 version of Meilisearch has been released, our company is also preparing to migrate the search system to Meilisearch. We also have a great demand for this function. We hope Meilisearch can support the most basic fixed item function as soon as possible

**Comment by MY-kkk** [comment]
A very good product that can help you purchase the necessary products, recommended for use

**Comment by Jengro777** [comment]
Has this been supported?

https://blog.meilisearch.com/promoted-search-results-with-react-instantsearch/?utm_campaign=oss&utm_source=productboard&utm_medium=portal

**Reply by curquiza** [reply]
Hello @Avey777 

What do you mean by "supported"?
Following the steps of this article should work.
If not, please let us know 😊
